p2p/server.go

```diff
@@ -403,7 +406,18 @@ func (srv *Server) Stop() {
 	}
 	close(srv.quit)
 	srv.lock.Unlock()
-	srv.loopWG.Wait()
+
+	stopChan := make(chan struct{})
+	go func() {
```

Comment:
`defer close(stopChan)`, This should not be a problem, but I think we should make a clean code.